dtfs-central-api/src/repositories/tfm-users-repo/tfm-users.repo.ts

```diff
@@ -0,0 +1,23 @@
+import { Collection, WithoutId, ObjectId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS, TfmUser } from '@ukef/dtfs2-common';
+import { mongoDbClient } from '../../drivers/db-client';
+
+export class TfmUsersRepo {
+  /**
+   * Gets the tfm users collection
+   * @returns The collection
+   */
+  private static async getCollection(): Promise<Collection<WithoutId<TfmUser>>> {
+    return mongoDbClient.getCollection(MONGO_DB_COLLECTIONS.TFM_USERS);
+  }
+
+  /**
+   * Finds a user by id
+   * @param id - The user id
+   * @returns The found user
+   */
+  public static async findOneUserById(id: string | ObjectId): Promise<TfmUser | null> {
+    const collection = await this.getCollection();
+    return await collection.findOne({ _id: new ObjectId(id) });
```

Comment:
I don't see a reason to enforce the argument to be `ObjectId` as that requires each caller of this method to use database related logic. In this case, the argument we supply from the controller will always be a `string` as it comes from JSON, so it makes no sense to have to inject the mongo dependency into the controller so that it can call the repo. Instead, it makes more sense to have all the database logic abstracted away to the repo and allow the repo caller to supply either an `ObjectId` or a `string`

Reply:
Resolving as `return await` has been removed. I agree that it's helpful for this function to consume either an `ObjectId` or `string` and simplify consumption.